sdk/storage/azfile/share/models.go

```diff
@@ -0,0 +1,149 @@
+type DeleteOptions struct {
+	// Specifies the option include to delete the base share and all of its snapshots.
+	DeleteSnapshots *DeleteSnapshotsOptionType
+	// TODO: Should snapshot be removed from the option bag
+	// The snapshot parameter is an opaque DateTime value that, when present, specifies the share snapshot to query.
+	Snapshot *string
+	// LeaseAccessConditions contains optional parameters to access leased entity.
+	LeaseAccessConditions *LeaseAccessConditions
+}
+
+// LeaseAccessConditions contains optional parameters to access leased entity.
+type LeaseAccessConditions = generated.LeaseAccessConditions
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+// RestoreOptions contains the optional parameters for the Client.Restore method.
+type RestoreOptions struct {
+	// placeholder for future options
+}
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+// GetPropertiesOptions contains the optional parameters for the Client.GetProperties method.
+type GetPropertiesOptions struct {
+	// TODO: Should snapshot be removed from the option bag
+	// The snapshot parameter is an opaque DateTime value that, when present, specifies the share snapshot to query.
+	Snapshot *string
+	// LeaseAccessConditions contains optional parameters to access leased entity.
+	LeaseAccessConditions *LeaseAccessConditions
+}
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+// SetPropertiesOptions contains the optional parameters for the Client.SetProperties method.
+type SetPropertiesOptions struct {
+	// Specifies the access tier of the share.
+	AccessTier *AccessTier
+	// Specifies the maximum size of the share, in gigabytes.
+	Quota *int32
+	// Root squash to set on the share. Only valid for NFS shares.
+	RootSquash *RootSquash
+	// LeaseAccessConditions contains optional parameters to access leased entity.
+	LeaseAccessConditions *LeaseAccessConditions
+}
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+// CreateSnapshotOptions contains the optional parameters for the Client.CreateSnapshot method.
+type CreateSnapshotOptions struct {
+	// A name-value pair to associate with a file storage object.
+	Metadata map[string]*string
+}
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+// GetAccessPolicyOptions contains the optional parameters for the Client.GetAccessPolicy method.
+type GetAccessPolicyOptions struct {
+	// LeaseAccessConditions contains optional parameters to access leased entity.
+	LeaseAccessConditions *LeaseAccessConditions
+}
+
+// SignedIdentifier - Signed identifier.
+type SignedIdentifier = generated.SignedIdentifier
```

Comment:
`Id` -> `ID` please change all instances of this in the codebase

Reply:
All instances of `ID` are already capitalized.